include/swift/AST/Module.h

```diff
@@ -844,20 +853,26 @@ class SourceFile final : public FileUnit {
 
     /// This source file has access to testable declarations in the imported
     /// module.
-    Testable = 0x2
+    Testable = 0x2,
+
+    /// This source file has access to private declarations in the imported
+    /// module.
+    PrivateImport = 0x4,
   };
 
   /// \see ImportFlags
   using ImportOptions = OptionSet<ImportFlags>;
 
+  typedef std::pair<ImportOptions, StringRef> ImportOptionsAndFilename;
 private:
   std::unique_ptr<LookupCache> Cache;
   LookupCache &getCache() const;
 
   /// This is the list of modules that are imported by this module.
   ///
   /// This is filled in by the Name Binding phase.
-  ArrayRef<std::pair<ModuleDecl::ImportedModule, ImportOptions>> Imports;
+  ArrayRef<std::pair<ModuleDecl::ImportedModule, ImportOptionsAndFilename>>
```

Comment:
A pair where the second type is again a pair is a bit too much IMHO. Can you just define a struct with three fields instead?